### PR TITLE
add statement about the number of promises `all` can handle

### PR DESCRIPTION
**Pull request by fitsyu**
hopefully this might provide extra useful information on solution for issue #109.

### CODE REVIEW

**Comment by fitsyu** [comment]
Hi, I signed it already. 
Thank you

File: g3doc/index.md [hunk 1]
```diff
@@ -1072,6 +1072,19 @@ method on `NSArray`, which often comes handy, along with other similar
 [functional operators](https://github.com/google/functional-objc) that
 Objective-C lacks.
 
+
+Note: The number of promises `all` can handle is 4. If you need more than 4,
```
Comment (shoumikhin) [inline g3doc/index.md]: Worth to mention that limitation appears only in Swift and only for promises of different types. Btw, for the [Any](text_features_size) operator the current limit is 3, may like to mention that in a corresponding docs section.

Comment (fitsyu) [reply g3doc/index.md]: Sure, it is worthy. I'll add it.  

I see that the swift-only and different promise types case has been mentioned in the previous paragraph.  

> **In Swift**, the all operator also allows passing promises of **heterogeneous types**. For this case, the returned promise will be resolved with a tuple containing the values of the input promises in the same order they were provided.

This paragraph looks great to put the limitation information.
And some lines of code seem helpful to give an example on how to solve this limitation issue. 
So it'll be added to where it was. 
Let me know if this is not fit.

Here is my checklist.
To add:

 - [ ] Swift Only
 - [ ] Applicable only to heterogeneous promises
 - [ ] Note for `any` limitation

File: g3doc/index.md [hunk 2]
```diff
@@ -1188,6 +1201,8 @@ methods on `NSArray`, which often comes handy, along with other similar
 [functional operators](https://github.com/google/functional-objc) that
 Objective-C lacks.
 
+Note: The number of promises that `any` can handle is 3.
```
Comment (shoumikhin) [inline g3doc/index.md]: Can you phrase the comment here similarly to what you had for all?
In Swift, the number of promises of heterogeneous types that `any` can handle is 3.

**Comment by shoumikhin** [comment]
Apologies, looks like this got missed and stayed here for too long :(
Can you do the last refinement to push it in asap?
Thanks!

File: g3doc/index.md (hunk 2 above)
Comment (fitsyu) [reply g3doc/index.md]: yes I can.
I'll be happy to do it.

**Comment by fitsyu** [comment]
No problem. I'll do it. Glad if it can be of help :)

**Comment by temrich** [comment]
@fitsyu Thank you for the PR!

**Comment by fitsyu** [comment]
> @fitsyu Thank you for the PR!

you're welcome!

**Pull request merged**